ghcide/src/Development/IDE/LSP/Notifications.hs

```diff
@@ -127,6 +128,12 @@ descriptor recorder plId = (defaultPluginDescriptor plId) { pluginNotificationHa
         setSomethingModified (VFSModified vfs) ide [toKey GetClientSettings emptyFilePath] "config change"
 
   , mkPluginNotificationHandler LSP.SMethod_Initialized $ \ide _ _ _ -> do
+
+      let params = ConfigurationParams [ConfigurationItem Nothing (Just "haskell.formattingProvider")]
+      void $ sendRequest LSP.SMethod_WorkspaceConfiguration params $ \res -> do
+        liftIO $ logWarning (ideLogger ide) $ "zltest: " <> (Text.pack $ show res)
+        pure ()
+
```

Comment:
you probably want to block waiting for the response to SMethod_WorkspaceConfiguration here?

Reply:
I think we don't need to. We have a default config, so we should always be able to operate and just update the config dynamically when we get the response, as we would if we got `didChangeConfiguration`.